### PR TITLE
@86epx9xt3 #step3 of BG info Items and TableView

### PR DESCRIPTION
**Pull request by acostales08**

### CODE REVIEW

File: packages/core-library/system/app/internal/blocks/Hub/Settings/SettingsManagement/steps/content/simulator/steps/content/casestudy/CaseStudySummary/component/Items/ItemContent.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+      return {
+        title,
+        id: index + 1,
+        content: <Items content={[item]} />,
+      };
+    });
+  };
+
+  const validQuestionnaires =
```
Comment (JMSevilla) [inline packages/core-library/system/app/internal/blocks/Hub/Settings/SettingsManagement/steps/content/simulator/steps/content/casestudy/CaseStudySummary/component/Items/ItemContent.tsx:25]: Can we instead use this kind of structure here

const VALID_QUESTION_TYPES = ["DDC", "SATA", "MRSN"];

const validQuestionnaires = values.questionnaires?.filter(
  (item): item is QuestionnaireItem => VALID_QUESTION_TYPES.includes(item.questionType)
) || [];

File: packages/core-library/system/app/internal/blocks/Hub/Settings/SettingsManagement/steps/content/simulator/steps/content/casestudy/CaseStudySummary/component/Items/items.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
+import { Box, Checkbox, Typography } from "@mui/material";
+import NearMeIcon from "@mui/icons-material/NearMe";
+import { DDCquestion } from "./DDCQuestion";
+import {
+  DDCAnswerOption,
+  QuestionnaireItem,
+} from "../../../../../../../../../../../../../types";
+import { useSanitizedInputs } from "../../../../../../../../../../../../../../../../hooks/useSanitizeInputs";
+
+export const Items: React.FC<{ content: QuestionnaireItem[] }> = ({
```
Comment (JMSevilla) [inline packages/core-library/system/app/internal/blocks/Hub/Settings/SettingsManagement/steps/content/simulator/steps/content/casestudy/CaseStudySummary/component/Items/items.tsx:10]: Kindly follow this code improvement 

use this constant to replace your sx that looks like repeating it self in other component 
const wordWrapStyles: SxProps<Theme> = {
    "& *": {
      margin: 0,
      padding: 0,
      lineHeight: 1.5,
      wordBreak: "break-word",
    },
  };
  
  
  don't forget to always follow `Separation of concerns`
  
  const renderQuestionType = (
  data: QuestionnaireItem,
  purifyInputs: (input: string) => string | TrustedHTML
) => {
  if (data.questionType === "DDC") {
    return (
      <DDCquestion
        ddcData={{
          answers: data.answers as DDCAnswerOption[],
          itemStem: data.itemStem,
        }}
      />
    );
  } else {
    return (
      <Typography
        sx={{
          "& *": {
            margin: 0,
            padding: 0,
            lineHeight: 1.5,
            wordBreak: "break-word",
          },
        }}
        dangerouslySetInnerHTML={{
          __html: purifyInputs(data.itemStem) as TrustedHTML,
        }}
      />
    );
  }
};

// Helper function to render labels for SATA and MRSN question types
const renderQuestionTypeLabel = (data: QuestionnaireItem) => {
  if (data.questionType === "SATA") {
    return (
      <Typography marginTop="14px" fontSize="16px" color="#999999" fontWeight="700">
        Select All That Apply
      </Typography>
    );
  } else if (data.questionType === "MRSN") {
    return (
      <Typography marginTop="14px" fontSize="16px" color="#999999" fontWeight="700">
        {`Select ${data.maxAnswer} That Apply`}
      </Typography>
    );
  }
  return null; // No label for other types
};

// Helper component to render answers
const AnswerList: React.FC<{ answers: Answer[] }> = ({ answers }) => {
  return (
    <Box marginTop="10px">
      {answers.map((answer) => (
        <Box display="flex" alignItems="center" paddingX="10px" key={answer.id}>
          <Checkbox disabled checked={answer.answerKey} />
          <Typography fontSize="16px">{answer.answer}</Typography>
        </Box>
      ))}
    </Box>
  );
};

File: packages/core-library/system/app/internal/blocks/Hub/Settings/SettingsManagement/steps/content/simulator/steps/content/casestudy/CaseStudySummary/component/TableView.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  Box,
+  Typography,
+} from "@mui/material";
+import { ContainedCaseStudyQuestionType } from "../../../../../types";
+
+interface Props {
+  data: Partial<ContainedCaseStudyQuestionType>;
+}
+
+export const TableView: React.FC<Props> = ({ data }) => {
```
Comment (JMSevilla) [inline packages/core-library/system/app/internal/blocks/Hub/Settings/SettingsManagement/steps/content/simulator/steps/content/casestudy/CaseStudySummary/component/TableView.tsx:18]: Can we instead use our `DataTable` component here again you created another component which we already have.

**Review by JMSevilla — CHANGES_REQUESTED** [review]

File: packages/core-library/system/app/internal/blocks/Hub/Settings/SettingsManagement/steps/content/simulator/steps/content/casestudy/CaseStudySummary/component/Items/items.tsx [hunk 4]
```diff
@@ -1,18 +1,74 @@
...
 import NearMeIcon from "@mui/icons-material/NearMe";
 import { DDCquestion } from "./DDCQuestion";
 import {
+  AnswerOption,
   DDCAnswerOption,
   QuestionnaireItem,
 } from "../../../../../../../../../../../../../types";
 import { useSanitizedInputs } from "../../../../../../../../../../../../../../../../hooks/useSanitizeInputs";
 
+export const wordWrapStyles: SxProps = {
```
Comment (JMSevilla) [inline packages/core-library/system/app/internal/blocks/Hub/Settings/SettingsManagement/steps/content/simulator/steps/content/casestudy/CaseStudySummary/component/Items/items.tsx:11]: If I'm not mistaken we do have a hook to place the styling for each component, coordinate with sam or jb

**Review by JMSevilla — CHANGES_REQUESTED** [review]

**Review by JMSevilla — APPROVED** [review]

**Review by DropitupQwerty — APPROVED** [review]

**Pull request merged**